### PR TITLE
Support R-package with cmake build and fix installation instructions

### PR DESCRIPTION
**Pull request by leezu**
## Description ##
Support R-package with cmake build and fix installation instructions

Also: Fixes https://github.com/apache/incubator-mxnet/issues/17103

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [X] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [X] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [X] Support building R-package with cmake built `libmxnet.so` 
- [X] Fix R-package installation instructions

### CODE REVIEW

**Comment by szha** [comment]
@hetong007 could you help verify this?

File: docs/static_site/src/pages/get_started/ubuntu_setup.md [hunk 1]
```diff
@@ -465,51 +451,37 @@ Or you can go through a manual process described next.
...
+**Step 1** Install R, cran-devtools and roxygen2
 
-**Step 3** Install OpenCV.
+If you are on Ubuntu 19.04 or higher:
 
-*MXNet* uses [OpenCV](https://opencv.org/) for efficient image loading and augmentation operations.
 ```bash
-$ sudo apt-get install -y libopencv-dev
+$ sudo apt-get update
+$ sudo apt-get install -y r-base-core r-cran-devtools r-cran-roxygen2
```
Comment (hetong007) [inline docs/static_site/src/pages/get_started/ubuntu_setup.md]: would you please hint me on what's the issue for `roxygen2` on ubuntu 19.04?

Comment (leezu) [reply docs/static_site/src/pages/get_started/ubuntu_setup.md]: It's not included in the ubuntu repo prior to ubuntu 19.04. So users on older systems need to install it manually.

**Comment by leezu** [comment]
I added a few more fixes to this PR. The current R-package was broken since #7273 #7389 due to missing linker arguments to the OpenCV dependency. It worked only due to some lucky circumstances such as order of dependency declaration in `libmxnet.so` that do not hold for CMake compiled version.

In particular, I updated the `Makevars` to fix the issue.

**Pull request merged**